pontoon/checks/utils/compare.py

```diff
@@ -0,0 +1,183 @@
+from collections import (
+    namedtuple,
+)
+
+from compare_locales.checks import getChecker
+from compare_locales.parser import (
+    FluentParser,
+)
+
+CommentEntity = namedtuple(
+    'Comment', (
+        'all',
+    )
+)
+
+File = namedtuple(
+    'File', (
+        'file',
+        'locale',
+    )
+)
+
+# Because we can't pass the context to all entities passed to compare locales,
+# we have to create our equivalents of compare-locale's internal classes.
+ComparePropertiesEntity = namedtuple(
+    'ComparePropertiesEntity',
+    (
+        'key',
+        'val',
+
+        # We'll remove these fields at some point, currently they're required because of the current
+        # implementation property files in compare-locales.
+        'raw_val',
+        'pre_comment',
+    )
+)
+CompareDTDEntity = namedtuple(
+    'CompareDTDEntity',
+    (
+        'key',
+        'val',
+        'raw_val',
+        'pre_comment',
+        'all',
+    )
+)
+CompareFluentEntity = namedtuple(
+    'FluentCompareEntity',
+    (
+        'entry',
+    )
+)
+
+
+DTD_ENTITY_TMPL = '<!ENTITY %s \"%s\">'
+
+
+class UnsupportedResourceTypeError(Exception):
+    """Raise if compare-locales doesn't support given resource-type."""
+    pass
+
+
+def cast_to_compare_locales(resource_ext, entity, string, plural_form):
+    """
+    Cast a Pontoon's translation object into Entities supported by `compare-locales`.
+
+    :arg basestring resource_ext: extension of a resource.
+    :arg pontoon.base.models.Entity entity: Source entity
+    :arg basestring string: a translation
+    :arg pontoon.base.models.Locale locale: Locale of a translation
+    :arg int plural_form: plural form of a translation
+    :return: source entity and translation entity that will be compatible a compare-locales checker.
+        Type of those entities depends on the resource_ext.
+    """
+    if resource_ext == '.properties':
+        if plural_form:
```

Comment:
This doesn't make sense for `.properties`, plural is gettext only, right?

Reply:
I'll double verify that and return to you.